### ISSUE TITLE
Possible to extract variables from Expression?

### ISSUE THREAD

**Issue by herrytco** [fkleon/math-expressions#73]
For my project, I need to parse expressions and get a list of used symbols in that expression. This would look something like this:

```
String input = "x+y*z+3";
Expression e = Parser().parse(input);

List<String> symbols = e.symbols; // ["x", "y", "z"]
```

Is that possible with the current state? I couldn't find the function on my own.

**Comment by fkleon** [comment]
Hi @herrytco thanks for creating this issue. It's currently not possible to get a list of symbols/variables directly from the expression tree.

**Comment by herrytco** [comment]
Thank you for the answer. Than I fall back to my current strategy to do the following: 

1. try to evaluate the expression
2. catch all errors related to unbound variables
3. set the variable to a random value
4. repeat 2-3 until the expression is evaluated
5. return the list of symbols found
6. profit

**Comment by VittorioParagallo** [comment]
any plan to implement this feature?

**Comment by fkleon** [comment]
> any plan to implement this feature?

No, but feel free to open a PR if you're wanting to contribute this.
